implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -310,9 +313,12 @@ private void send(List<String> metricLines) {
                     .withPlainText(body)
                     .send()
                     .onSuccess(response -> handleSuccess(metricLines.size(), response))
-                    .onError(response -> logger.error("Failed metric ingestion: Error Code={}, Response Body={}", response.code(), response.body()));
+                    .onError(response -> logger.error("Failed metric ingestion: Error Code={}, Response Body={}",
+                        response.code(),
+                        StringUtils.truncate(response.body(), LOG_RESPONSE_BODY_TRUNCATION_LIMIT)));
         } catch (Throwable throwable) {
-            logger.error("Failed metric ingestion: " + throwable.getMessage(), throwable);
+            logger.error("Failed metric ingestion: " + throwable);
+            logger.debug(throwable);
```

Comment:
Is the idea of this coming from a user?
I usually like seeing the stacktrace if an error happened, it makes it easy where the error is coming from. Logging it out on `DEBUG` level defeats that purpose, you need to know where the log event is coming from so that you can see (and read the code) that a stacktrace is available on a lower level.

Reply:
Also, the `DEBUG` event will not contain a message so it will look weird: it will not tell you what happened and nothing guarantees that these events will be next to (or close to) each other in the logs.

If we decide that we don't want the stacktrace on `ERROR` level (I would advise against it), I would state that the user sees this stacktrace because metric ingestion failed and there is another log event about it on `ERROR` level.

Reply:
There is [`WarnThenDebugLogger`](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-commons/src/main/java/io/micrometer/common/util/internal/logging/WarnThenDebugLogger.java) that logs at warn level at first and then logs at debug level for the rest of the events.

Reply:
I changed it to always print the message as a warning and use the WarnThenDebugLogger for the stack including a reference to the other warning message. This is under the assumption that it will usually be the same error if it would occur multiple times and that the overall context is clear from the log message prefix.